src/meta/processors/admin/HBProcessor.cpp

```diff
@@ -114,5 +115,14 @@ void HBProcessor::process(const cpp2::HBReq& req) {
   onFinished();
 }
 
+void HBProcessor::setLeaderInfo() {
+  auto leaderRet = kvstore_->partLeader(kDefaultSpaceId, kDefaultPartId);
+  if (ok(leaderRet)) {
+    resp_.leader_ref() = toThriftHost(nebula::value(leaderRet));
+  } else {
+    resp_.code_ref() = nebula::error(leaderRet);
```

Comment:
will this replace the "E_MACHINE_NOT_FOUND"? Which one we will want to return to the user?

Reply:
In most cases, this will not happen. when the leader is not found, the priority is higher than machine not found.